tests/components/mqtt/test_light_template.py

```diff
@@ -320,9 +326,284 @@
+                    "{{ blue|d }}",
+                    "command_off_template": "off",
+                    "state_template": '{{ value.split(",")[0] }}',
+                    "brightness_template": '{{ value.split(",")[1] }}',
+                    "color_temp_template": '{{ value.split(",")[2] }}',
+                    "white_value_template": '{{ value.split(",")[3] }}',
+                    "red_template": '{{ value.split(",")[4].' 'split("-")[0] }}',
+                    "green_template": '{{ value.split(",")[4].' 'split("-")[1] }}',
+                    "blue_template": '{{ value.split(",")[4].' 'split("-")[2] }}',
+                    "effect_template": '{{ value.split(",")[5] }}',
+                }
+            },
+        )
+
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+    assert not state.attributes.get("brightness")
+    assert not state.attributes.get("hs_color")
+    assert not state.attributes.get("effect")
+    assert not state.attributes.get("color_temp")
+    assert not state.attributes.get("white_value")
+    assert not state.attributes.get(ATTR_ASSUMED_STATE)
+
+    await common.async_turn_off(hass, "light.test")
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "off", 0, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+
+    await common.async_turn_on(hass, "light.test")
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,,,,--", 0, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+
+    # Set color_temp
+    await common.async_turn_on(hass, "light.test", color_temp=70)
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,,70,,--", 0, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+    assert not state.attributes.get("color_temp")
+
+    # Set full brightness
+    await common.async_turn_on(hass, "light.test", brightness=255)
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,255,,,--", 0, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+    assert not state.attributes.get("brightness")
+
+    # Full brightness - no scaling of RGB values sent over MQTT
+    await common.async_turn_on(
+        hass, "light.test", rgb_color=[255, 128, 0], white_value=80
+    )
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,,,80,255-128-0", 0, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+    assert not state.attributes.get("white_value")
+    assert not state.attributes.get("rgb_color")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5199018)